sklearn/utils/tests/test_param_validation.py

```diff
@@ -403,11 +403,23 @@ def test_validate_params_match_error():
     def func(a, b):
         pass
 
-    match = r"The parameter constraints .* do not match the parameters to validate"
+    match = r"The parameter constraints .* contain unexpected parameters {'c'}"
```

Comment:
I think the docstring of this test can be updated to reflect the new behavior as well.